sam-sql-database/src/agents/sql_database/sql_database_agent_component.py

```diff
@@ -302,28 +319,21 @@ def get_agent_summary(self):
 
         if self.data_description:
             description += f"Data Description:\n{self.data_description}\n"
-        else:
+        
+        # Extract table information if schema exists
+        if isinstance(self.detailed_schema, str):
             try:
-                # Only try to parse as YAML if we have a string that might be YAML
-                if isinstance(self.detailed_schema, str):
-                    schema_dict = yaml.safe_load(self.detailed_schema)
-                    if isinstance(schema_dict, dict):
-                        tables = list(schema_dict.keys())
-                        if tables:
-                            description += f"Contains {len(tables)} tables: {', '.join(tables)}\n"
-                        else:
-                            description += "No tables found in database.\n"
-                    else:
-                        description += "Schema information not available in a valid format"
-                else:
-                    description += "Schema information not available in a valid format"
+                schema_dict = yaml.safe_load(self.detailed_schema)
+                if isinstance(schema_dict, dict) and schema_dict:
+                    tables = list(schema_dict.keys())
+                    description += f"Contains {len(tables)} tables: {', '.join(tables)}\n"
```

Comment:
It feels like we should build this once at init time and save the full description. I think we are doing all this work every 15 seconds even though it never changes